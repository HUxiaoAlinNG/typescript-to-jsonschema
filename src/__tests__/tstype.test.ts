import * as path from 'path';

import genTypeSchema from '../index';

const file = path.join(__dirname, `../../jest/tstype.ts`);
genTypeSchema.genJsonDataFormFile(file);

const getSchema = (type: string) => {
  const schema = genTypeSchema.getJsonSchema(file, type);
  return schema;
};

test('type类型_1', () => {
  expect(getSchema('Type_1')).toMatchSnapshot();
});
test('type类型_2', () => {
  expect(getSchema('Type_2')).toMatchSnapshot();
});
test('type类型_3', () => {
  expect(getSchema('Type_3')).toMatchSnapshot();
});
test('type类型_4', () => {
  expect(getSchema('Type_4')).toMatchSnapshot();
});
test('type类型_5', () => {
  expect(getSchema('Type_5')).toMatchSnapshot();
});
test('type类型_6', () => {
  expect(getSchema('Type_6')).toMatchSnapshot();
});
test('type类型_7', () => {
  expect(getSchema('Type_7')).toMatchSnapshot();
});
test('type类型_8', () => {
  expect(getSchema('Type_8')).toMatchSnapshot();
});
test('type类型_9', () => {
  expect(getSchema('Type_9')).toMatchSnapshot();
});
test('type类型_10', () => {
  expect(getSchema('Type_10')).toMatchSnapshot();
});
test('type类型_11', () => {
  expect(getSchema('Type_11')).toMatchSnapshot();
});
test('type类型_数组_1', () => {
  expect(getSchema('Type_12')).toMatchSnapshot();
});
test('type类型_空对象_1', () => {
  expect(getSchema('Type_13')).toMatchSnapshot();
});

test('type类型_数组_2', () => {
  expect(getSchema('Type_14')).toMatchSnapshot();
});

