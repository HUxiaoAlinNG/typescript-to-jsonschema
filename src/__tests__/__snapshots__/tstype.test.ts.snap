// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`type类型_1 1`] = `
Object {
  "description": "单行备注",
  "example": "1",
  "type": "number",
}
`;

exports[`type类型_2 1`] = `
Object {
  "description": "多行备注",
  "example": "'zane'",
  "type": "string",
}
`;

exports[`type类型_3 1`] = `
Object {
  "anyOf": Array [
    Object {
      "type": "string",
    },
    Object {
      "type": "number",
    },
  ],
}
`;

exports[`type类型_4 1`] = `
Object {
  "anyOf": Array [
    Object {
      "type": "string",
    },
    Object {
      "type": "number",
    },
    Object {
      "enum": Array [
        5,
        true,
      ],
    },
  ],
}
`;

exports[`type类型_5 1`] = `
Object {
  "additionalProperties": false,
  "definitions": Object {},
  "properties": Object {
    "other1": Object {
      "type": "string",
    },
  },
  "required": Array [
    "other1",
  ],
  "type": "object",
}
`;

exports[`type类型_6 1`] = `
Object {
  "anyOf": Array [
    Object {
      "$ref": "#/definitions/AAA",
    },
    Object {
      "type": "number",
    },
  ],
  "definitions": Object {
    "AAA": Object {
      "additionalProperties": false,
      "properties": Object {
        "other1": Object {
          "type": "string",
        },
      },
      "required": Array [
        "other1",
      ],
      "type": "object",
    },
  },
}
`;

exports[`type类型_7 1`] = `
Object {
  "anyOf": Array [
    Object {
      "$ref": "#/definitions/AAA",
    },
    Object {
      "$ref": "#/definitions/Type_1",
    },
  ],
  "definitions": Object {
    "AAA": Object {
      "additionalProperties": false,
      "properties": Object {
        "other1": Object {
          "type": "string",
        },
      },
      "required": Array [
        "other1",
      ],
      "type": "object",
    },
    "Type_1": Object {
      "description": "单行备注",
      "example": "1",
      "type": "number",
    },
  },
}
`;

exports[`type类型_8 1`] = `
Object {
  "definitions": Object {
    "AAA": Object {
      "additionalProperties": false,
      "properties": Object {
        "other1": Object {
          "type": "string",
        },
      },
      "required": Array [
        "other1",
      ],
      "type": "object",
    },
    "Type_1": Object {
      "description": "单行备注",
      "example": "1",
      "type": "number",
    },
  },
  "items": Object {
    "anyOf": Array [
      Object {
        "$ref": "#/definitions/AAA",
      },
      Object {
        "$ref": "#/definitions/Type_1",
      },
    ],
  },
  "type": "array",
}
`;

exports[`type类型_9 1`] = `
Object {
  "properties": Object {
    "a": Object {
      "type": "number",
    },
    "c": Object {
      "type": "boolean",
    },
  },
  "required": Array [
    "a",
    "c",
  ],
  "type": "object",
}
`;

exports[`type类型_10 1`] = `
Object {
  "definitions": Object {},
  "enum": Array [
    1,
    2,
    3,
  ],
  "type": "number",
}
`;

exports[`type类型_11 1`] = `
Object {
  "definitions": Object {
    "Label": Object {
      "enum": Array [
        1,
        2,
        3,
      ],
      "type": "number",
    },
  },
  "properties": Object {
    "name": Object {
      "$ref": "#/definitions/Label",
    },
  },
  "required": Array [
    "name",
  ],
  "type": "object",
}
`;

exports[`type类型_数组_1 1`] = `
Object {
  "definitions": Object {
    "AAA": Object {
      "additionalProperties": false,
      "properties": Object {
        "other1": Object {
          "type": "string",
        },
      },
      "required": Array [
        "other1",
      ],
      "type": "object",
    },
  },
  "items": Object {
    "$ref": "#/definitions/AAA",
  },
  "type": "array",
}
`;

exports[`type类型_空对象_1 1`] = `
Object {
  "type": "object",
}
`;
